flink-connector-debezium/src/main/java/com/ververica/cdc/debezium/DebeziumSourceFunction.java

```diff
@@ -390,6 +390,9 @@ public void run(SourceContext<T> sourceContext) throws Exception {
         // and https://debezium.io/blog/2018/03/16/note-on-database-history-topic-configuration/
         properties.setProperty("database.history", determineDatabase().getCanonicalName());
 
+        // we need this in order not to lose any transaction during snapshot to streaming switch
+        properties.setProperty("internal.log.mining.transaction.snapshot.boundary.mode", "all");
```

Comment:
This config option is just worked in `oracle-cdc`.But in flink cdc this config is useless when snapshot by flink cdc without using debezium snapshot.

Reply:
Oh, my bad, I misplaced it. Transferred it to `OracleSource#build()`. But, unfortunately, I don't understand second part of your comment. Catching pending transactions during snapshot finish works only in debezium and if incremental snapshot mode is enabled, then this functionality won't work, right. But implementing such behavior in incremental sources is beyond the scope of this pr.

Reply:
I means `internal.log.mining.transaction.snapshot.boundary.mode` this config will not worked in parallelismSnapshot. And I think this config option can also keeped.

Reply:
Yes, now this config is enabled by default in non incremental oracle source